src/logger.cpp

```diff
@@ -44,28 +48,106 @@ Logger::~Logger()
     delete mPimpl;
 }
 
-void Logger::removeAllErrors()
+size_t Logger::errorCount() const
+{
+    return mPimpl->mErrors.size();
+}
+
+IssuePtr Logger::error(size_t index) const
+{
+    IssuePtr issue = nullptr;
+    if ((index < mPimpl->mErrors.size()) && (mPimpl->mErrors.at(index) < mPimpl->mIssues.size())) {
```

Comment:
Why do we need to test for `mPimpl->mErrors.at(index) < mPimpl->mIssues.size()`? Surely, indexes held `mPimpl->mErrors` should be valid?

Reply:
Same for `warning()` and `hint()`.

Reply:
You'd hope so ... but better safe than segfaulty?

Reply:
Hmm, if you feel the need to have that test then, to me, it means that you don't trust your code in the first place. So, either you trust it and you don't need that test, or you don't trust it and review it.

Reply:
Done in commit 9ff91db2.